plugins/parquet/parquetembed.cpp

```diff
@@ -262,6 +267,29 @@ void divide_row_groups(const IThorActivityContext *activityCtx, __int64 totalRow
     }
 }
 
+/**
+ * @brief Reads selected columns from the current Table in the Parquet file.
+ *
+ * @param currTable The index of the Table to read columns from.
+ * @return __int64 The number of rows in the current Table.
+ */
+__int64 ParquetReader::readColumns(__int64 currTable)
```

Comment:
It's not a concern at the moment because it is only called in one place. I think it would be best to make the method private because it probably won't ever need to be called outside of that one codepath where expectedRecord is used to decide which method to call for reading the columns.

Reply:
There were some additional methods that should not be called anywhere else. I have made them private as well.